server/src/main/java/com/hedera/block/server/persistence/storage/read/BlockAsDirReaderBuilder.java

```diff
@@ -16,37 +16,40 @@
 
 package com.hedera.block.server.persistence.storage.read;
 
-import com.hedera.block.server.persistence.storage.FileUtils;
+import static com.hedera.block.common.utils.FileUtilities.DEFAULT_DIR_PERMISSIONS;
+
 import com.hedera.block.server.persistence.storage.PersistenceStorageConfig;
 import com.hedera.hapi.block.stream.Block;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import java.nio.file.attribute.FileAttribute;
 import java.nio.file.attribute.PosixFilePermission;
+import java.util.Objects;
 import java.util.Set;
 
 /**
  * Use builder methods to create a {@link BlockReader} to read blocks from storage.
  *
  * <p>When a block reader is created, it will provide access to read blocks from storage.
  */
-public class BlockAsDirReaderBuilder {
-
-    private FileAttribute<Set<PosixFilePermission>> filePerms = FileUtils.defaultPerms;
+public final class BlockAsDirReaderBuilder {
     private final PersistenceStorageConfig config;
+    private FileAttribute<Set<PosixFilePermission>> filePerms = DEFAULT_DIR_PERMISSIONS;
```

Comment:
This is bad practice.
1. Copying a constant to a class attribute is an anti-pattern
1. Using a remote default constant when an overload exists to apply defaults _internally_ to the library is an anti-pattern.

As far as the way this builder is setup, the _default_ should be `null`.
That way, the BlockAsDirReader it creates _either_ calls the commons overload that takes no permissions, _or_ calls the overload _with_ permissions (both file _and_ folder, which _must be different_) and has a configured value for that.

Reply:
Same as above comment

Reply:
✅ Done.